### PR TITLE
Refactor RustCycle constructors

### PR DESCRIPTION
**Pull request by kylecarow**
- [x] Remove `RustCycle::new` and `RustCycle.__new__`
- [x] to_/from_ PyDict methods
- [x] Impl To/From HashMap (ignores `name` field)

### CODE REVIEW

File: rust/fastsim-core/src/cycle.rs [hunk 1]
```diff
@@ -490,14 +495,38 @@ impl RustCycleCache {
...
+                Array::from_vec(value.extract()?)
+            } else {
+                Array::default(cyc_len)
+            },
+            name: PyAny::get_item(&dict, "name").and_then(String::extract).unwrap_or_default(),
+            orphaned: false,
+        })
+    }
+
+    // TODO: okay to rename `get_cyc_dict` in Python and Rust to `to_dict`?
```
Comment (calbaker) [inline rust/fastsim-core/src/cycle.rs]: @kylecarow , yes, do this rename

File: rust/fastsim-cli/src/bin/fastsim-cli.rs [hunk 2]
```diff
@@ -226,13 +227,14 @@ pub fn main() -> anyhow::Result<()> {
         RustCycle::from_file(adopt_hd_string)
     } else {
         //TODO? use pathbuff to string, for robustness
```
Comment (calbaker) [inline rust/fastsim-cli/src/bin/fastsim-cli.rs:229]: @kylecarow , is this comment deprecated here?

Comment (kylecarow) [reply rust/fastsim-cli/src/bin/fastsim-cli.rs:229]: Not sure, that's leftover from Andrew Guttman. I think we could delete the TODO.

**Comment by kylecarow** [comment]
@calbaker This is ready to merge BTW

**Review by calbaker — APPROVED** [review]

**Pull request merged**